lib/demos/reference/motion_demo_shared_y_axis_transition.dart

```diff
@@ -0,0 +1,167 @@
+import 'dart:math';
+import 'package:flutter/material.dart';
+import 'package:animations/animations.dart';
+import 'package:gallery/l10n/gallery_localizations.dart';
+
+// BEGIN sharedYAxisTransitionDemo
+
+class SharedYAxisTransitionDemo extends StatefulWidget {
+  const SharedYAxisTransitionDemo({Key key}) : super(key: key);
+
+  @override
+  _SharedYAxisTransitionDemoState createState() =>
+      _SharedYAxisTransitionDemoState();
+}
+
+class _SharedYAxisTransitionDemoState extends State<SharedYAxisTransitionDemo>
+    with SingleTickerProviderStateMixin {
+  bool _isAlphabetical = false;
+  AnimationController _controller;
+
+  @override
+  void initState() {
+    super.initState();
+    _controller = AnimationController(
+      duration: const Duration(milliseconds: 300),
+      vsync: this,
+    );
+  }
+
+  final _recentList = ListView(
+    // Adding [UniqueKey] to make sure the widget rebuilds when transitioning.
+    key: UniqueKey(),
+    children: [
+      ...List.generate(10, (index) => _AlbumTile((index + 1).toString()))
+    ],
+  );
+
+  final _alphabeticalList = ListView(
+    // Adding [UniqueKey] to make sure the widget rebuilds when transitioning.
+    key: UniqueKey(),
+    children: _alphabet.map((letter) => _AlbumTile(letter)).toList(),
+  );
+
+  static const _alphabet = [
+    'A',
+    'B',
+    'C',
+    'D',
+    'E',
+    'F',
+    'G',
+    'H',
+    'I',
+    'J',
+  ];
+
+  @override
+  Widget build(BuildContext context) {
+    final localizations = GalleryLocalizations.of(context);
+
+    return Scaffold(
+      appBar: AppBar(
+        automaticallyImplyLeading: false,
+        title: Text(localizations.demoSharedYAxisTitle),
+      ),
+      body: Column(
+        children: [
+          const SizedBox(height: 5),
+          Row(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+            children: [
+              Padding(
+                padding: const EdgeInsets.only(left: 15),
+                child: Text(localizations.demoSharedYAxisAlbumCount),
+              ),
+              Padding(
+                padding: const EdgeInsets.only(right: 7),
+                child: Row(
+                  children: [
+                    Text(_isAlphabetical
+                        ? localizations.demoSharedYAxisAlphabeticalSortTitle
+                        : localizations.demoSharedYAxisRecentSortTitle),
+                    InkWell(
+                      customBorder: const CircleBorder(),
+                      child: RotationTransition(
+                        turns: Tween(begin: 0.0, end: 1.0).animate(_controller),
+                        child: const Icon(Icons.arrow_drop_down),
+                      ),
+                      onTap: () {
+                        setState(() {
```

Comment:
does this need to call setState? calling controller.animateTo might animate without needed to call setState first